src/InArrayValidator.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace ipl\Validator;
+
+use ipl\I18n\Translation;
+
+/**
+ * Validate if specific single or multiple values exist in an array
+ */
+class InArrayValidator extends BaseValidator
+{
+    use Translation;
+
+    /** @var array The array */
+    protected $haystack = [];
+
+    /** @var bool Whether the types of the needle in the haystack should also match */
+    protected $strict = false;
+
+    /**
+     * Create a new InArray validator
+     *
+     * **Optional options:**
+     *
+     * * `haystack`: (`array`) The array
+     * * `strict`: (`bool`) Whether the types of the needle in the haystack should also match, default `false`
+     *
+     * @param array $options
+     */
+    public function __construct(array $options = [])
+    {
+        if (isset($options['haystack'])) {
+            $this->setHaystack($options['haystack']);
+        }
+
+        $this->setStrict($options['strict'] ?? false);
+    }
+
+    /**
+     * Get the haystack
+     *
+     * @return array
+     */
+    public function getHaystack(): array
+    {
+        return $this->haystack;
+    }
+
+    /**
+     * Set the haystack
+     *
+     * @param array $haystack
+     *
+     * @return $this
+     */
+    public function setHaystack(array $haystack): self
+    {
+        $this->haystack = $haystack;
+
+        return $this;
+    }
+
+    /**
+     * Whether the types of the needle in the haystack should also match
+     *
+     * @return bool
+     */
+    public function isStrict(): bool
+    {
+        return $this->strict;
+    }
+
+    /**
+     * Set whether the types of the needle in the haystack should also match
+     *
+     * @param bool $strict
+     *
+     * @return $this
+     */
+    public function setStrict(bool $strict = true): self
+    {
+        $this->strict = $strict;
+
+        return $this;
+    }
+
+    public function isValid($value)
+    {
+        // Multiple isValid() calls must not stack validation messages
+        $this->clearMessages();
+
+        $notInArray = $this->diffValues((array) $value);
+
+        if (empty($notInArray)) {
+            return true;
+        }
+
+        $this->addMessage(sprintf(
+            $this->translatePlural(
+                "%s was not found in the haystack",
+                "%s were not found in the haystack",
+                count($notInArray)
+            ),
+            implode(', ', $notInArray)
+        ));
+
+        return false;
+    }
+
+    /**
+     * Computes the difference of the given values with the help of the haystack
+     *
+     * @param array $values
+     *
+     * @return array Values that was not found in the haystack
+     */
+    public function diffValues(array $values = []): array
```

Comment:
I would call this method something like `findInvalid`. `diffValues` is far too general for what it actually does, and you need PHPDoc to explain that.